include/openssl/dh.h

```diff
@@ -160,7 +160,7 @@ int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key);
 void DH_clear_flags(DH *dh, int flags);
 int DH_test_flags(const DH *dh, int flags);
 void DH_set_flags(DH *dh, int flags);
-ENGINE *DH_get0_engine(DH *d);
+const ENGINE *DH_get0_engine(const DH *d);
```

Comment:
It's been pointed out that there's not much you can do with a const-engine, since code typically needs to up the refcount of engines when they pass them around. Possibly we simply shouldn't bother with `ENGINE`.